### DISCUSSION TITLE
Building dnGrep: Which version of Visual Studio to use?

### DISCUSSION THREAD

**Discussion by fade2gray** [dnGrep/dnGrep#685] [Q&A] [answered]
I've tried opening `dnGREP.WPF.sln` with 2019 and 2022, both present me with…
<details>
<summary>See image (<i>click to expand</i>)</summary>

![wix](https://user-images.githubusercontent.com/1332009/173110137-f8903e09-0b22-4d1a-b9c3-9a4a641472cf.PNG)
</details>

…followed by various errors and warnings.
<details>
<summary>Error list</summary>

- Error		NuGet Package restore failed for project TestLocalizedStrings: Unable to find version '2.2.6' of package 'AlphaFS'.
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\: Package 'AlphaFS.2.2.6' is not found on source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\'.
. Please see Error List window for detailed warnings and errors.				
- Error	NU1101	Unable to find package xunit. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	Tests	V:\dnGrep\3.0.64.0\f2g\build\VS2022\Tests\Tests.csproj	1	
- Error	NU1101	Unable to find package xunit.runner.visualstudio. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	Tests	V:\dnGrep\3.0.64.0\f2g\build\VS2022\Tests\Tests.csproj	1	
- Error	NU1101	Unable to find package UTF.Unknown. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.Engines	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.Engines\dnGREP.Engines.csproj	1	
- Error	NU1102	Unable to find package Newtonsoft.Json with version (>= 13.0.1)
  \- Found 1 version(s) in Microsoft Visual Studio Offline Packages [ Nearest version: 9.0.1 ]	dnGREP.Engines	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.Engines\dnGREP.Engines.csproj	1	
- Error	NU1101	Unable to find package NLog. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.Engines	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.Engines\dnGREP.Engines.csproj	1	
- Error	NU1101	Unable to find package AlphaFS. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.Engines	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.Engines\dnGREP.Engines.csproj	1	
- Error	NU1101	Unable to find package DocumentFormat.OpenXml. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.OpenXmlEngine	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.OpenXmlEngine\dnGREP.OpenXmlEngine.csproj	1	
- Error	NU1101	Unable to find package ExcelNumberFormat. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.OpenXmlEngine	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.OpenXmlEngine\dnGREP.OpenXmlEngine.csproj	1	
- Error	NU1101	Unable to find package ExcelDataReader. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.OpenXmlEngine	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.OpenXmlEngine\dnGREP.OpenXmlEngine.csproj	1	
- Error	NU1101	Unable to find package StrongNamer. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	dnGREP.WPF	V:\dnGrep\3.0.64.0\f2g\build\VS2022\dnGREP.WPF\dnGREP.WPF.csproj	1	
- Error	NU1101	Unable to find package Microsoft.SourceLink.GitHub. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages	ICSharpCode.AvalonEdit	V:\dnGrep\3.0.64.0\f2g\build\VS2022\ICSharpCode.AvalonEdit\ICSharpCode.AvalonEdit.csproj	1	
</details>

The yellow warning triangles disapeared after installing WiX Toolset via `Menu>Tools>Get Tools and Features`, where I can find v3.x in `C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\WiX\v3.x` (no idea where to find v4).
![WixTooset](https://user-images.githubusercontent.com/1332009/173136062-69b1eb91-c460-48c9-a96c-6e8661648b65.PNG)

But when looking at `dnGREP.Setup\dnGREP.Setup.wixproj`, at line 5, I see `C:\Program Files (x86)\WiX Toolset v3.8`, which does not exist.

**Comment by doug24** [comment, marked as answer]
I currently use VS 2022, but VS 2019 should still work.  I just set up a new pc, so I just did this with VS 2022.   I installed:
WIX 3.11.2 (the latest) [from here](https://wixtoolset.org/releases/)
WIX Visual Studio Extension - for the latest VS 2022 release you need the [1.0.0.20 prerelease](https://github.com/wixtoolset/VisualStudioExtension/releases/tag/v1.0.0.20) for a bug fix.

For me, Visual Studio downloaded all the NuGet packages automatically when I opened the project.  Maybe your NuGet settings are not correct?

![capture-000086](https://user-images.githubusercontent.com/17227248/173148661-eec6056a-9704-4632-b465-e257947a14e6.png)
![capture-000088](https://user-images.githubusercontent.com/17227248/173148689-f1a7609b-eb42-4be6-aa95-dc070675321c.png)

**Comment by fade2gray** [comment]
Getting somewhere now, thanks.
